documentation/source/conf.py

```diff
@@ -334,6 +334,11 @@
 spelling_ignore_wiki_words = False
 spelling_show_suggestions = True
 
-# -- Setup for inheritance_diagram directive which uses graphviz ---------------
+# -- Extra setup for inheritance_diagram directive which uses graphviz ---------
 
 graphviz_output_format = 'svg'
+
+# -- Extra setup for towncrier -------------------------------------------------
+# see also https://towncrier.readthedocs.io/en/actual-freaking-docs/
+
+subprocess.call(['towncrier', '--yes'], cwd='../..')
```

Comment:
Does `--draft` help at all?

Reply:
Yeah, it sounds like you want:
```
in_progress_notes = subprocess.check_output(['towncrier', '--draft'])
with open('source/generated/master-notes.rst', 'w') as f:
    f.write(in_progress_notes )
```
and then add a `.. include source/generated/master-notes.rst` in the release notes.

Reply:
Ok, that looks good.
Let me try to write down then what to do at release time assuming this change (feel free to edit this text directly):

> * ``cd documentation/source/``
> * insert the contents of ``generated/master-notes.rst`` into ``release_notes.rst`` below the existing ``.. include``
> * move the heading (with the version number this release is for) from above the ``.. include`` to below it
> * add a new heading with the next version number above ``.. include``
> * run ``git rm newsfragments/[0-9]+.*.rst``

Anything I forgot?

Reply:
I don't think you need any of that. Isn't the step at release time just run `towncrier` by itself?

Reply:
> move the heading (with the version number this release is for) from above the .. include to below it

This is solved by just having `conf.py` output the heading as part of `master-notes.rst`

Reply:
Oh right, we just leave the special comment for towncrier in, next to the include.

Reply:
Bingo - the special comment goes below the include, and is exercised only during release